### PR TITLE
Error screen when pairing with wrong metamask wallet

### PR DESCRIPTION
**Pull request by dakshsilence**

### CODE REVIEW

File: lib/screens/scanner_screen.dart [hunk 1]
```diff
@@ -190,6 +191,15 @@ class _ScannerScreenState extends State<ScannerScreen> {
           }),
         ),
       );
+    } else if (error is StateError && error.toString().contains('INVALID_BACKUP_DATA')) {
+      Navigator.of(context).push(
+        MaterialPageRoute(
+          builder: (context) => WrongMetaMaskWalletForRecoveryScreen(onPress: () {
+            _updateScannerState(ScannerState.scanning);
```
Comment (noahsilencelab) [inline lib/screens/scanner_screen.dart]: whenever we use `_resetScannerController()` we will update ScannerState to `scanning`, should we move that update logic in to `_resetScannerController()` as well?

Comment (dakshsilence) [reply lib/screens/scanner_screen.dart]: Both are independent right now. We can create new function calling `_resetScannerController() ` and update ScannerState to `scanning`.

Comment (noahsilencelab) [reply lib/screens/scanner_screen.dart]: yes, that's what I meant. Update ScannerState to scanning inside `_resetScannerController`

**Review by noahsilencelab — APPROVED** [review]
lgtm

**Pull request merged**